framework/src/base/LazyCoupleable.C

```diff
@@ -0,0 +1,64 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "LazyCoupleable.h"
+#include "FEProblem.h"
+#include "MooseVariable.h"
+#include "InputParameters.h"
+#include "MooseObject.h"
+
+#include "libmesh/simple_range.h"
+
+LazyCoupleable::LazyCoupleable(const MooseObject * moose_object)
+  : _l_parameters(moose_object->parameters()),
+    _l_name(_l_parameters.get<std::string>("_object_name"))
+{
+  for (const auto & var_name :
+       as_range(std::make_pair(_l_parameters.coupledVarsBegin(), _l_parameters.coupledVarsEnd())))
+    _coupled_var_numbers[var_name] = libmesh_make_unique<unsigned int>(0);
+}
+
+LazyCoupleable::~LazyCoupleable() {}
```

Comment:
We omit empty destructors now, right?

Reply:
Yes and No, you need them for the BASE CLASS, but not any derived classes. However, looking at this again. I don't intend for this class to be derived from so I'm going to remove it.